scylla-cdc/src/consumer.rs

```diff
@@ -0,0 +1,485 @@
+use num_enum::TryFromPrimitive;
+use scylla::frame::response::result::{ColumnSpec, CqlValue, Row};
+use std::collections::HashMap;
+
+pub trait Consumer {
+    fn consume_cdc(&mut self, data: CDCRow);
+}
+
+pub trait ConsumerFactory {
+    fn new_consumer(&self) -> Box<dyn Consumer>;
+}
+
+#[derive(Debug, Eq, PartialEq, TryFromPrimitive)]
+#[repr(i8)]
+pub enum OperationType {
+    PreImage,
+    RowUpdate,
+    RowInsert,
+    RowDelete,
+    PartitionDelete,
+    RowRangeDelInclLeft,
+    RowRangeDelExclLeft,
+    RowRangeDelInclRight,
+    RowRangeDelExclRight,
+    PostImage,
+}
+
+pub struct CDCRowSchema {
+    // The usize values are indices of given values in the Row.columns vector.
+    pub(crate) stream_id: usize,
+    pub(crate) time: usize,
+    pub(crate) batch_seq_no: usize,
+    pub(crate) end_of_batch: usize,
+    pub(crate) operation: usize,
+    pub(crate) ttl: usize,
+
+    // These HashMaps map names of columns in the observed table to matching columns in the CDC table.
+    // The usize value is an index of the column in an internal vector inside of the CDCRow struct.
+
+    // Maps name of a column to a matching column in the CDC table.
+    pub(crate) mapping: HashMap<String, usize>,
+    // Maps name of a column to a column that tells if value in this column was deleted.
+    pub(crate) deleted_mapping: HashMap<String, usize>,
+    // Maps name of a collection column to a column that tells
+    // if elements from this collection were deleted.
+    pub(crate) deleted_el_mapping: HashMap<String, usize>,
+}
+
+const STREAM_ID_NAME: &str = "cdc$stream_id";
+const TIME_NAME: &str = "cdc$time";
+const BATCH_SEQ_NO_NAME: &str = "cdc$batch_seq_no";
+const END_OF_BATCH_NAME: &str = "cdc$end_of_batch";
+const OPERATION_NAME: &str = "cdc$operation";
+const TTL_NAME: &str = "cdc$ttl";
+const IS_DELETED_PREFIX: &str = "cdc$deleted_";
+const ARE_ELEMENTS_DELETED_PREFIX: &str = "cdc$deleted_elements_";
+
+impl CDCRowSchema {
+    pub fn new(specs: &[ColumnSpec]) -> CDCRowSchema {
+        // [TODO]:  This function is not usable now.
+        //          The driver does not provide the vector of ColumnSpec for every query yet,
+        //          and therefore we can't map the result
+        //          to something other than a hardcoded tuple.
```

Comment:
Please remove this comment. I suspect nobody will bother to remove it later, and it can only cause confusion. I'm not sure it provides any value.